apoc.custom.function error on startup

## Expected Behavior (Mandatory)

No error on startup for apoc.custom.*.   The function does not show up with a call to  apoc.custom.list()
## Actual Behavior (Mandatory)

Don't error.

## How to Reproduce the Problem

1. Take attached dump, unzip and load into a 4.2.5 Neo4j EE database (I used Neo4j Desktop to create this db)
1. Install apoc
1. Start db

### Simple Dataset (where it's possibile)

[contract-tracing-name-idx-neo4j-2021-04-26T214357.dump.zip](https://github.com/neo4j-contrib/neo4j-apoc-procedures/files/6380772/contract-tracing-name-idx-neo4j-2021-04-26T214357.dump.zip)

### Steps (Mandatory)

1. Start db
2. Review logs/neo4j.log

## Screenshots (where it's possibile)

Initial relevant section of neo4j.log
```
2021-04-26 21:39:02.627+0000 INFO  Starting...
2021-04-26 21:39:05.482+0000 INFO  ======== Neo4j 4.2.5 ========
2021-04-26 21:39:10.533+0000 ERROR [neo4j/99376483] Could not register function: custom.debugQry(level = OFF :: STRING?, threshold = 0s :: STRING?, logsLogLevel = NONE :: STRING?) :: (STRING?)
with: CALL dbms.setConfigValue('dbms.logs.query.enabled', $level)
 CALL dbms.setConfigValue('dbms.logs.query.page_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.threshold', $threshold)
 CALL dbms.setConfigValue('dbms.logs.query.allocation_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.runtime_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.parameter_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.time_logging_enabled', 'TRUE')

 CALL dbms.setConfigValue('dbms.logs.debug.level', $logsLogLevel)

 CALL dbms.setConfigValue('dbms.track_query_allocation', 'TRUE')
 CALL dbms.setConfigValue('dbms.track_query_cpu_time', 'TRUE')
 RETURN [{level: $level}, {threshold: $threshold}, {logsLogLevel: $logsLogLevel}] AS res
 single result true
java.lang.UnsupportedOperationException: null
	at java.util.AbstractCollection.add(AbstractCollection.java:267) ~[?:?]
	at apoc.custom.CypherProceduresHandler.registerFunction(CypherProceduresHandler.java:410) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at apoc.custom.CypherProceduresHandler$UserFunctionDescriptor.register(CypherProceduresHandler.java:686) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at apoc.custom.CypherProceduresHandler.lambda$restoreProceduresAndFunctions$2(CypherProceduresHandler.java:212) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?]
	at apoc.custom.CypherProceduresHandler.restoreProceduresAndFunctions(CypherProceduresHandler.java:211) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at apoc.custom.CypherProceduresHandler.available(CypherProceduresHandler.java:120) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at org.neo4j.internal.helpers.Listeners.notifySingleListener(Listeners.java:126) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.internal.helpers.Listeners.notify(Listeners.java:83) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailabilityGuard.fulfill(DatabaseAvailabilityGuard.java:119) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailability.start(DatabaseAvailability.java:61) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.database.Database.start(Database.java:511) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.dbms.database.AbstractDatabaseManager.startDatabase(AbstractDatabaseManager.java:187) ~[neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.forSingleDatabase(MultiDatabaseManager.java:141) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.startDatabase(MultiDatabaseManager.java:119) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransitionAction(Transition.java:101) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransition(Transition.java:88) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:355) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$12(DbmsReconciler.java:324) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.namedJob(DbmsReconciler.java:335) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitions(DbmsReconciler.java:325) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$10(DbmsReconciler.java:310) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) [?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2021-04-26 21:39:12.006+0000 INFO  Called db.clearQueryCaches(): Query cache already empty.
2021-04-26 21:39:19.478+0000 INFO  Sending metrics to CSV file at /Users/danflavin/Library/Application Support/com.Neo4j.Relate/Data/dbmss/dbms-9c3de7fb-c391-461c-8b77-2be30636f488/metrics
2021-04-26 21:39:19.529+0000 INFO  Bolt enabled on localhost:7687.
2021-04-26 21:39:20.651+0000 INFO  Remote interface available at http://localhost:7474/
2021-04-26 21:39:20.652+0000 INFO  Started.
2021-04-26 21:39:20.927+0000 INFO  Neo4j Server shutdown initiated by request
2021-04-26 21:39:20.927+0000 INFO  Stopping...
2021-04-26 21:39:26.809+0000 INFO  Stopped.
2021-04-26 21:39:45.584+0000 INFO  Starting...
2021-04-26 21:39:48.426+0000 INFO  ======== Neo4j 4.2.5 ========
2021-04-26 21:39:51.752+0000 ERROR [neo4j/99376483] Could not register function: custom.debugQry(level = OFF :: STRING?, threshold = 0s :: STRING?, logsLogLevel = NONE :: STRING?) :: (STRING?)
with: CALL dbms.setConfigValue('dbms.logs.query.enabled', $level)
 CALL dbms.setConfigValue('dbms.logs.query.page_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.threshold', $threshold)
 CALL dbms.setConfigValue('dbms.logs.query.allocation_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.runtime_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.parameter_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.time_logging_enabled', 'TRUE')

 CALL dbms.setConfigValue('dbms.logs.debug.level', $logsLogLevel)

 CALL dbms.setConfigValue('dbms.track_query_allocation', 'TRUE')
 CALL dbms.setConfigValue('dbms.track_query_cpu_time', 'TRUE')
 RETURN [{level: $level}, {threshold: $threshold}, {logsLogLevel: $logsLogLevel}] AS res
 single result true
java.lang.UnsupportedOperationException: null
	at java.util.AbstractCollection.add(AbstractCollection.java:267) ~[?:?]
	at apoc.custom.CypherProceduresHandler.registerFunction(CypherProceduresHandler.java:414) ~[apoc-4.2.0.2.jar:4.2.0.2]
	at apoc.custom.CypherProceduresHandler$UserFunctionDescriptor.register(CypherProceduresHandler.java:692) ~[apoc-4.2.0.2.jar:4.2.0.2]
	at apoc.custom.CypherProceduresHandler.lambda$restoreProceduresAndFunctions$2(CypherProceduresHandler.java:214) ~[apoc-4.2.0.2.jar:4.2.0.2]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?]
	at apoc.custom.CypherProceduresHandler.restoreProceduresAndFunctions(CypherProceduresHandler.java:213) ~[apoc-4.2.0.2.jar:4.2.0.2]
	at apoc.custom.CypherProceduresHandler.available(CypherProceduresHandler.java:120) ~[apoc-4.2.0.2.jar:4.2.0.2]
	at org.neo4j.internal.helpers.Listeners.notifySingleListener(Listeners.java:126) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.internal.helpers.Listeners.notify(Listeners.java:83) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailabilityGuard.fulfill(DatabaseAvailabilityGuard.java:119) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailability.start(DatabaseAvailability.java:61) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.database.Database.start(Database.java:511) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.dbms.database.AbstractDatabaseManager.startDatabase(AbstractDatabaseManager.java:187) ~[neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.forSingleDatabase(MultiDatabaseManager.java:141) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.startDatabase(MultiDatabaseManager.java:119) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransitionAction(Transition.java:101) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransition(Transition.java:88) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:355) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$12(DbmsReconciler.java:324) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.namedJob(DbmsReconciler.java:335) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitions(DbmsReconciler.java:325) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$10(DbmsReconciler.java:310) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) [?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2021-04-26 21:39:53.125+0000 INFO  Called db.clearQueryCaches(): Query cache already empty.
2021-04-26 21:40:07.389+0000 ERROR Failed to start Neo4j on dbms.connector.http.listen_address, a socket address. If missing port or hostname it is acquired from dbms.default_listen_address.
java.lang.RuntimeException: Error starting Neo4j database server at /Users/danflavin/Library/Application Support/com.Neo4j.Relate/Data/dbmss/dbms-3944ce61-8fa4-4f02-b8d9-472601496be9/data/databases
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:207) ~[neo4j-4.2.5.jar:4.2.5]
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.build(DatabaseManagementServiceFactory.java:163) ~[neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.server.enterprise.EnterpriseManagementServiceFactory.createManagementService(EnterpriseManagementServiceFactory.java:38) ~[neo4j-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.server.enterprise.EnterpriseBootstrapper.createNeo(EnterpriseBootstrapper.java:20) ~[neo4j-enterprise-4.2.5.jar:4.2.5]
	at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:134) [neo4j-4.2.5.jar:4.2.5]
	at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:90) [neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.server.enterprise.EnterpriseEntryPoint.main(EnterpriseEntryPoint.java:25) [neo4j-enterprise-4.2.5.jar:4.2.5]
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.procedure.impl.GlobalProceduresRegistry@6537ac' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to register procedure, because the name `apoc.generate.ba` is already in use.".
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:463) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:198) ~[neo4j-4.2.5.jar:4.2.5]
	... 6 more
Caused by: org.neo4j.internal.kernel.api.exceptions.ProcedureException: Unable to register procedure, because the name `apoc.generate.ba` is already in use.
	at org.neo4j.procedure.impl.ProcedureRegistry.register(ProcedureRegistry.java:93) ~[neo4j-procedure-4.2.5.jar:4.2.5]
	at org.neo4j.procedure.impl.GlobalProceduresRegistry.register(GlobalProceduresRegistry.java:154) ~[neo4j-procedure-4.2.5.jar:4.2.5]
	at org.neo4j.procedure.impl.GlobalProceduresRegistry.register(GlobalProceduresRegistry.java:94) ~[neo4j-procedure-4.2.5.jar:4.2.5]
	at org.neo4j.procedure.impl.GlobalProceduresRegistry.start(GlobalProceduresRegistry.java:386) ~[neo4j-procedure-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:198) ~[neo4j-4.2.5.jar:4.2.5]
	... 6 more
2021-04-26 21:40:07.390+0000 INFO  Neo4j Server shutdown initiated by request
2021-04-26 21:40:07.390+0000 INFO  Stopped.
2021-04-26 21:41:55.168+0000 INFO  Starting...
2021-04-26 21:41:57.916+0000 INFO  ======== Neo4j 4.2.5 ========
2021-04-26 21:42:01.253+0000 ERROR [neo4j/99376483] Could not register function: custom.debugQry(level = OFF :: STRING?, threshold = 0s :: STRING?, logsLogLevel = NONE :: STRING?) :: (STRING?)
with: CALL dbms.setConfigValue('dbms.logs.query.enabled', $level)
 CALL dbms.setConfigValue('dbms.logs.query.page_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.threshold', $threshold)
 CALL dbms.setConfigValue('dbms.logs.query.allocation_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.runtime_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.parameter_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.time_logging_enabled', 'TRUE')

 CALL dbms.setConfigValue('dbms.logs.debug.level', $logsLogLevel)

 CALL dbms.setConfigValue('dbms.track_query_allocation', 'TRUE')
 CALL dbms.setConfigValue('dbms.track_query_cpu_time', 'TRUE')
 RETURN [{level: $level}, {threshold: $threshold}, {logsLogLevel: $logsLogLevel}] AS res
 single result true
java.lang.UnsupportedOperationException: null
	at java.util.AbstractCollection.add(AbstractCollection.java:267) ~[?:?]
	at apoc.custom.CypherProceduresHandler.registerFunction(CypherProceduresHandler.java:410) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at apoc.custom.CypherProceduresHandler$UserFunctionDescriptor.register(CypherProceduresHandler.java:686) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at apoc.custom.CypherProceduresHandler.lambda$restoreProceduresAndFunctions$2(CypherProceduresHandler.java:212) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?]
	at apoc.custom.CypherProceduresHandler.restoreProceduresAndFunctions(CypherProceduresHandler.java:211) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at apoc.custom.CypherProceduresHandler.available(CypherProceduresHandler.java:120) ~[apoc-4.2.0.0.jar:4.2.0.0]
	at org.neo4j.internal.helpers.Listeners.notifySingleListener(Listeners.java:126) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.internal.helpers.Listeners.notify(Listeners.java:83) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailabilityGuard.fulfill(DatabaseAvailabilityGuard.java:119) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailability.start(DatabaseAvailability.java:61) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.database.Database.start(Database.java:511) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.dbms.database.AbstractDatabaseManager.startDatabase(AbstractDatabaseManager.java:187) ~[neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.forSingleDatabase(MultiDatabaseManager.java:141) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.startDatabase(MultiDatabaseManager.java:119) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransitionAction(Transition.java:101) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransition(Transition.java:88) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:355) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$12(DbmsReconciler.java:324) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.namedJob(DbmsReconciler.java:335) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitions(DbmsReconciler.java:325) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$10(DbmsReconciler.java:310) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) [?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2021-04-26 21:42:02.657+0000 INFO  Called db.clearQueryCaches(): Query cache already empty.
2021-04-26 21:42:10.392+0000 INFO  Sending metrics to CSV file at /Users/danflavin/Library/Application Support/com.Neo4j.Relate/Data/dbmss/dbms-3944ce61-8fa4-4f02-b8d9-472601496be9/metrics
2021-04-26 21:42:10.425+0000 INFO  Bolt enabled on localhost:7687.
2021-04-26 21:42:11.532+0000 INFO  Remote interface available at http://localhost:7474/
2021-04-26 21:42:11.533+0000 INFO  Started
```

## Specifications (Mandatory)

Code used to create function:
```
CALL apoc.custom.asFunction('debugQry',
'CALL dbms.setConfigValue(\'dbms.logs.query.enabled\', $level)
 CALL dbms.setConfigValue(\'dbms.logs.query.page_logging_enabled\', \'TRUE\')
 CALL dbms.setConfigValue(\'dbms.logs.query.threshold\', $threshold)
 CALL dbms.setConfigValue(\'dbms.logs.query.allocation_logging_enabled\', \'TRUE\')
 CALL dbms.setConfigValue(\'dbms.logs.query.runtime_logging_enabled\', \'TRUE\')
 CALL dbms.setConfigValue(\'dbms.logs.query.parameter_logging_enabled\', \'TRUE\')
 CALL dbms.setConfigValue(\'dbms.logs.query.time_logging_enabled\', \'TRUE\')

 CALL dbms.setConfigValue(\'dbms.logs.debug.level\', $logsLogLevel)

 CALL dbms.setConfigValue(\'dbms.track_query_allocation\', \'TRUE\')
 CALL dbms.setConfigValue(\'dbms.track_query_cpu_time\', \'TRUE\')
 RETURN [{level: $level}, {threshold: $threshold}, {logsLogLevel: $logsLogLevel}] AS res',
'RETURN res',
[['level', 'STRING', "'OFF'"], ['threshold', 'STRING', "'0s'"], ['logsLogLevel', 'STRING', "'NONE'"]],  TRUE, 'Set debug: RETURN custom.debugQry(\'VERBOSE\', \'0s\', \'INFO\')')
;
```

Currently used versions

### Versions

  - OS: mac osx Big Sur 11.2.3
  - Neo4j: 4.2.5
  - Neo4j-Apoc: 4.2.0.2

Just tried apoc 4.2.0.3 on a 4.2.5 db per the original problem definition and I still get the same error / unable to start db.  Hence not resolved by #1744 unfortunately

```
2021-05-04 19:17:35.299+0000 INFO  Starting...
2021-05-04 19:17:38.092+0000 INFO  ======== Neo4j 4.2.5 ========
2021-05-04 19:17:41.207+0000 ERROR [neo4j/99376483] Could not register function: custom.debugQry(level = OFF :: STRING?, threshold = 0s :: STRING?, logsLogLevel = NONE :: STRING?) :: (STRING?)
with: CALL dbms.setConfigValue('dbms.logs.query.enabled', $level)
 CALL dbms.setConfigValue('dbms.logs.query.page_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.threshold', $threshold)
 CALL dbms.setConfigValue('dbms.logs.query.allocation_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.runtime_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.parameter_logging_enabled', 'TRUE')
 CALL dbms.setConfigValue('dbms.logs.query.time_logging_enabled', 'TRUE')

 CALL dbms.setConfigValue('dbms.logs.debug.level', $logsLogLevel)

 CALL dbms.setConfigValue('dbms.track_query_allocation', 'TRUE')
 CALL dbms.setConfigValue('dbms.track_query_cpu_time', 'TRUE')
 RETURN [{level: $level}, {threshold: $threshold}, {logsLogLevel: $logsLogLevel}] AS res
 single result true
java.lang.UnsupportedOperationException: null
	at java.util.AbstractCollection.add(AbstractCollection.java:267) ~[?:?]
	at apoc.custom.CypherProceduresHandler.registerFunction(CypherProceduresHandler.java:412) ~[apoc-4.2.0.3.jar:4.2.0.3]
	at apoc.custom.CypherProceduresHandler$UserFunctionDescriptor.register(CypherProceduresHandler.java:694) ~[apoc-4.2.0.3.jar:4.2.0.3]
	at apoc.custom.CypherProceduresHandler.lambda$restoreProceduresAndFunctions$2(CypherProceduresHandler.java:213) ~[apoc-4.2.0.3.jar:4.2.0.3]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?]
	at apoc.custom.CypherProceduresHandler.restoreProceduresAndFunctions(CypherProceduresHandler.java:212) ~[apoc-4.2.0.3.jar:4.2.0.3]
	at apoc.custom.CypherProceduresHandler.available(CypherProceduresHandler.java:119) ~[apoc-4.2.0.3.jar:4.2.0.3]
	at org.neo4j.internal.helpers.Listeners.notifySingleListener(Listeners.java:126) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.internal.helpers.Listeners.notify(Listeners.java:83) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailabilityGuard.fulfill(DatabaseAvailabilityGuard.java:119) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.availability.DatabaseAvailability.start(DatabaseAvailability.java:61) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.database.Database.start(Database.java:511) ~[neo4j-kernel-4.2.5.jar:4.2.5]
	at org.neo4j.dbms.database.AbstractDatabaseManager.startDatabase(AbstractDatabaseManager.java:187) ~[neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.forSingleDatabase(MultiDatabaseManager.java:141) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.database.MultiDatabaseManager.startDatabase(MultiDatabaseManager.java:119) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransitionAction(Transition.java:101) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.Transition$Prepared.doTransition(Transition.java:88) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:355) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitionStep(DbmsReconciler.java:356) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$12(DbmsReconciler.java:324) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.namedJob(DbmsReconciler.java:335) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.doTransitions(DbmsReconciler.java:325) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.dbms.DbmsReconciler.lambda$doTransitions$10(DbmsReconciler.java:310) ~[neo4j-dbms-enterprise-4.2.5.jar:4.2.5]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) [?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2021-05-04 19:17:42.538+0000 INFO  Called db.clearQueryCaches(): Query cache already empty.
2021-05-04 19:17:48.574+0000 ERROR Failed to start Neo4j on dbms.connector.http.listen_address, a socket address. If missing port or hostname it is acquired from dbms.default_listen_address.
java.lang.RuntimeException: Error starting Neo4j database server at /Users/danflavin/Library/Application Support/com.Neo4j.Relate/Data/dbmss/dbms-3944ce61-8fa4-4f02-b8d9-472601496be9/data/databases
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:207) ~[neo4j-4.2.5.jar:4.2.5]
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.build(DatabaseManagementServiceFactory.java:163) ~[neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.server.enterprise.EnterpriseManagementServiceFactory.createManagementService(EnterpriseManagementServiceFactory.java:38) ~[neo4j-enterprise-4.2.5.jar:4.2.5]
	at com.neo4j.server.enterprise.EnterpriseBootstrapper.createNeo(EnterpriseBootstrapper.java:20) ~[neo4j-enterprise-4.2.5.jar:4.2.5]
	at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:134) [neo4j-4.2.5.jar:4.2.5]
	at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:90) [neo4j-4.2.5.jar:4.2.5]
	at com.neo4j.server.enterprise.EnterpriseEntryPoint.main(EnterpriseEntryPoint.java:25) [neo4j-enterprise-4.2.5.jar:4.2.5]
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.procedure.impl.GlobalProceduresRegistry@3b6c2be6' was successfully initialized, but failed to start. Please see the attached cause exception "Some jar procedure files (bloom-plugin-4.x-1.5.1.jar) are invalid, see log for details.".
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:463) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:198) ~[neo4j-4.2.5.jar:4.2.5]
	... 6 more
Caused by: java.util.zip.ZipException: Some jar procedure files (bloom-plugin-4.x-1.5.1.jar) are invalid, see log for details.
	at org.neo4j.procedure.impl.ProcedureJarLoader.loadProceduresFromDir(ProcedureJarLoader.java:82) ~[neo4j-procedure-4.2.5.jar:4.2.5]
	at org.neo4j.procedure.impl.GlobalProceduresRegistry.start(GlobalProceduresRegistry.java:383) ~[neo4j-procedure-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.2.5.jar:4.2.5]
	at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:198) ~[neo4j-4.2.5.jar:4.2.5]
	... 6 more
2021-05-04 19:17:48.575+0000 INFO  Neo4j Server shutdown initiated by request
2021-05-04 19:17:48.575+0000 INFO  Stopped.
```

--- a/full/src/main/java/apoc/custom/CypherProceduresHandler.java
+++ b/full/src/main/java/apoc/custom/CypherProceduresHandler.java
@@ -254,6 +254,7 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
             setLastUpdate(tx);
             registerFunction(signature, statement, forceSingle);
+            registeredUserFunctionSignatures.add(signature);
             return null;
         });
     }
@@ -272,6 +273,7 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
             node.setProperty(SystemPropertyKeys.mode.name(), signature.mode().name());
             setLastUpdate(tx);
             registerProcedure(signature, statement);
+            registeredProcedureSignatures.add(signature);
             return null;
         });
     }
@@ -361,9 +363,12 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
                     }
                 }
             }, true);
+//            System.out.println("CypherProceduresHandler.registerProcedure1");
             registeredProcedureSignatures.add(signature);
             return true;
         } catch (Exception e) {
+//            System.out.println("e.getCause().getMessage()" + e.getCause().getMessage());
+            System.out.println("e.getClass().getName()" + e.getClass().getName());
             log.error("Could not register procedure: " + signature.name() + " with " + statement + "\n accepting" + signature.inputSignature() + " resulting in " + signature.outputSignature() + " mode " + signature.mode(), e);
             return false;
         }
@@ -410,7 +415,7 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
                 }
             }, true);
-            registeredUserFunctionSignatures.add(signature);
+//            registeredUserFunctionSignatures.add(signature);
             return true;
         } catch (Exception e) {
             log.error("Could not register function: " + signature + "\nwith: " + statement + "\n single result " + forceSingle, e);

--- a/full/src/main/java/apoc/custom/CypherProceduresHandler.java
+++ b/full/src/main/java/apoc/custom/CypherProceduresHandler.java
@@ -98,8 +98,8 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
     private final JobScheduler jobScheduler;
     private long lastUpdate;
     private final ThrowingFunction<Context, Transaction, ProcedureException> transactionComponentFunction;
-    private Set<ProcedureSignature> registeredProcedureSignatures = emptySet();
-    private Set<UserFunctionSignature> registeredUserFunctionSignatures = emptySet();
+    private final Set<ProcedureSignature> registeredProcedureSignatures = Collections.synchronizedSet(new HashSet<>());
+    private final Set<UserFunctionSignature> registeredUserFunctionSignatures = Collections.synchronizedSet(new HashSet<>());
     private static Group REFRESH_GROUP = Group.STORAGE_MAINTENANCE;
     private JobHandle restoreProceduresHandle;
 
@@ -205,28 +205,23 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
     public void restoreProceduresAndFunctions() {
         lastUpdate = System.currentTimeMillis();
-        Set<ProcedureSignature> currentProcedureSignatures = Collections.synchronizedSet(new HashSet<>());
-        Set<UserFunctionSignature> currentUserFunctionSignatures = Collections.synchronizedSet(new HashSet<>());
+        Set<ProcedureSignature> currentProceduresToRemove = new HashSet<>(registeredProcedureSignatures);
+        Set<UserFunctionSignature> currentUserFunctionsToRemove = new HashSet<>(registeredUserFunctionSignatures);
 
         readSignatures().forEach(descriptor -> {
             descriptor.register();
             if (descriptor instanceof ProcedureDescriptor) {
                 ProcedureSignature signature = ((ProcedureDescriptor) descriptor).getSignature();
-                currentProcedureSignatures.add(signature);
-                registeredProcedureSignatures.remove(signature);
+                currentProceduresToRemove.remove(signature);
             } else {
                 UserFunctionSignature signature = ((UserFunctionDescriptor) descriptor).getSignature();
-                currentUserFunctionSignatures.add(signature);
-                registeredUserFunctionSignatures.remove(signature);
+                currentUserFunctionsToRemove.remove(signature);
             }
         });
 
         // de-register removed procs/functions
-        registeredProcedureSignatures.forEach(signature -> registerProcedure(signature, null));
-        registeredUserFunctionSignatures.forEach(signature -> registerFunction(signature, null, false));
-
-        registeredProcedureSignatures = currentProcedureSignatures;
-        registeredUserFunctionSignatures = currentUserFunctionSignatures;
+        currentProceduresToRemove.forEach(signature -> registerProcedure(signature, null));
+        currentUserFunctionsToRemove.forEach(signature -> registerFunction(signature, null, false));
 
         api.executeTransactionally("call db.clearQueryCaches()");
     }
@@ -341,10 +336,11 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
      */
     public boolean registerProcedure(ProcedureSignature signature, String statement) {
         try {
+            final boolean isStatementNull = statement == null;
             globalProceduresRegistry.register(new CallableProcedure.BasicProcedure(signature) {
                 @Override
                 public RawIterator<AnyValue[], ProcedureException> apply(org.neo4j.kernel.api.procedure.Context ctx, AnyValue[] input, ResourceTracker resourceTracker) throws ProcedureException {
-                    if (statement == null) {
+                    if (isStatementNull) {
                         final String error = String.format("There is no procedure with the name `%s` registered for this database instance. " +
                                 "Please ensure you've spelled the procedure name correctly and that the procedure is properly deployed.", signature.name());
                         throw new QueryExecutionException(error, null, "Neo.ClientError.Statement.SyntaxError");
@@ -363,6 +359,11 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
                     }
                 }
             }, true);
+            if (isStatementNull) {
+                registeredProcedureSignatures.remove(signature);
+            } else {
+                registeredProcedureSignatures.add(signature);
+            }
             return true;
         } catch (Exception e) {
             log.error("Could not register procedure: " + signature.name() + " with " + statement + "\n accepting" + signature.inputSignature() + " resulting in " + signature.outputSignature() + " mode " + signature.mode(), e);
@@ -372,10 +373,11 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
     public boolean registerFunction(UserFunctionSignature signature, String statement, boolean forceSingle) {
         try {
+            final boolean isStatementNull = statement == null;
             globalProceduresRegistry.register(new CallableUserFunction.BasicUserFunction(signature) {
                 @Override
                 public AnyValue apply(org.neo4j.kernel.api.procedure.Context ctx, AnyValue[] input) throws ProcedureException {
-                    if (statement == null) {
+                    if (isStatementNull) {
                         final String error = String.format("Unknown function '%s'", signature.name());
                         throw new QueryExecutionException(error, null, "Neo.ClientError.Statement.SyntaxError");
                     } else {
@@ -411,6 +413,11 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
                 }
             }, true);
+            if (isStatementNull) {
+                registeredUserFunctionSignatures.remove(signature);
+            } else {
+                registeredUserFunctionSignatures.add(signature);
+            }
             return true;
         } catch (Exception e) {
             log.error("Could not register function: " + signature + "\nwith: " + statement + "\n single result " + forceSingle, e);

--- a/full/src/main/java/apoc/custom/CypherProceduresHandler.java
+++ b/full/src/main/java/apoc/custom/CypherProceduresHandler.java
@@ -363,12 +363,8 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
                     }
                 }
             }, true);
-//            System.out.println("CypherProceduresHandler.registerProcedure1");
-            registeredProcedureSignatures.add(signature);
             return true;
         } catch (Exception e) {
-//            System.out.println("e.getCause().getMessage()" + e.getCause().getMessage());
-            System.out.println("e.getClass().getName()" + e.getClass().getName());
             log.error("Could not register procedure: " + signature.name() + " with " + statement + "\n accepting" + signature.inputSignature() + " resulting in " + signature.outputSignature() + " mode " + signature.mode(), e);
             return false;
         }
@@ -415,7 +411,6 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
                 }
             }, true);
-//            registeredUserFunctionSignatures.add(signature);
             return true;
         } catch (Exception e) {
             log.error("Could not register function: " + signature + "\nwith: " + statement + "\n single result " + forceSingle, e);

--- a/full/src/main/java/apoc/custom/CypherProceduresHandler.java
+++ b/full/src/main/java/apoc/custom/CypherProceduresHandler.java
@@ -249,7 +249,6 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
 
             setLastUpdate(tx);
             registerFunction(signature, statement, forceSingle);
-            registeredUserFunctionSignatures.add(signature);
             return null;
         });
     }
@@ -268,7 +267,6 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
             node.setProperty(SystemPropertyKeys.mode.name(), signature.mode().name());
             setLastUpdate(tx);
             registerProcedure(signature, statement);
-            registeredProcedureSignatures.add(signature);
             return null;
         });
     }

--- a/full/src/main/java/apoc/custom/CypherProceduresHandler.java
+++ b/full/src/main/java/apoc/custom/CypherProceduresHandler.java
@@ -618,7 +618,6 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
             ).stream().filter(n -> n.hasLabel(SystemLabels.Procedure)).forEach(node -> {
                 ProcedureDescriptor descriptor = procedureDescriptor(node);
                 registerProcedure(descriptor.getSignature(), null);
-                registeredProcedureSignatures.remove(descriptor.getSignature());
                 node.delete();
                 setLastUpdate(tx);
             });
@@ -636,7 +635,6 @@ public class CypherProceduresHandler extends LifecycleAdapter implements Availab
             ).stream().filter(n -> n.hasLabel(SystemLabels.Function)).forEach(node -> {
                 UserFunctionDescriptor descriptor = userFunctionDescriptor(node);
                 registerFunction(descriptor.getSignature(), null, false);
-                registeredUserFunctionSignatures.remove(descriptor.getSignature());
                 node.delete();
                 setLastUpdate(tx);
             });

--- a/full/src/test/java/apoc/custom/CypherProceduresStorageTest.java
+++ b/full/src/test/java/apoc/custom/CypherProceduresStorageTest.java
@@ -1,5 +1,6 @@
 package apoc.custom;
 
+import apoc.log.Neo4jLogStream;
 import apoc.util.TestUtil;
 import org.junit.Before;
 import org.junit.Rule;
@@ -9,14 +10,20 @@ import org.neo4j.configuration.GraphDatabaseSettings;
 import org.neo4j.dbms.api.DatabaseManagementService;
 import org.neo4j.graphdb.GraphDatabaseService;
 import org.neo4j.graphdb.Node;
+import org.neo4j.internal.helpers.collection.Iterators;
+import org.neo4j.logging.AssertableLogProvider;
+import org.neo4j.logging.Log;
 import org.neo4j.test.TestDatabaseManagementServiceBuilder;
 
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
 
+import static apoc.ApocConfig.apocConfig;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 
 /**
  * @author mh
@@ -30,9 +37,12 @@ public class CypherProceduresStorageTest {
     private GraphDatabaseService db;
     private DatabaseManagementService databaseManagementService;
 
+    public static AssertableLogProvider logProvider = new AssertableLogProvider(true);
+
     @Before
     public void setUp() throws Exception {
-        databaseManagementService = new TestDatabaseManagementServiceBuilder(STORE_DIR.getRoot()).build();
+        databaseManagementService = new TestDatabaseManagementServiceBuilder(STORE_DIR.getRoot())
+                .setUserLogProvider(logProvider).build();
         db = databaseManagementService.database(GraphDatabaseSettings.DEFAULT_DATABASE_NAME);
         TestUtil.registerProcedure(db, CypherProcedures.class);
     }
@@ -41,8 +51,8 @@ public class CypherProceduresStorageTest {
         databaseManagementService.shutdown();
         databaseManagementService = new TestDatabaseManagementServiceBuilder(STORE_DIR.getRoot()).build();
         db = databaseManagementService.database(GraphDatabaseSettings.DEFAULT_DATABASE_NAME);
-        assertTrue(db.isAvailable(1000));
-        TestUtil.registerProcedure(db, CypherProcedures.class);
+        assertTrue(db.isAvailable(60000));
+        TestUtil.registerProcedure(db, CypherProcedures.class, Neo4jLogStream.class);
     }
     @Test
     public void registerSimpleStatement() throws Exception {
@@ -193,5 +203,31 @@ public class CypherProceduresStorageTest {
         db.executeTransactionally("CALL apoc.custom.asFunction('n', 'MATCH (t:Target {value : $val}) RETURN t', 'NODE', [['val', 'INTEGER']])");
         restartDb();
         TestUtil.testCall(db, "RETURN custom.n(2) as row", (row) -> assertEquals(2L, ((Node) row.get("row")).getProperty("value")));
+    }
+
+    // TODO - METTERE LA PROCEDURA COMPLETA
+    @Test
+    public void testIssue1744() throws Exception {
+        db.executeTransactionally("CREATE (i:Target {value: 2});");
+        db.executeTransactionally("CALL apoc.custom.asProcedure('franco', 'RETURN \"1\" as resource', 'read', [['resource','STRING']]);");
+        restartDb();
+        TestUtil.testCall(db, "call custom.franco", (row) -> assertEquals("1", row.get("resource")));
+
+
+        final String serialize = logProvider.serialize();
+        assertFalse(serialize.contains("Could not register procedure"));
+//        final Log log = logProvider.getLog("");
+//         Could not register procedure:
+//        TestUtil.testResult(db, "call apoc.log.stream('neo4j.log')", (row) -> {
+//            final List<Map<String, Object>> maps = Iterators.asList(row);
+//            int intero = 0;
+//            while (row.hasNext()) {
+//                Map<String, Object> value = row.next();
+//                assertFalse(((String) value.get("line")).contains("Could not register procedure"));
+//                intero++;
+//                System.out.println("intero - " + intero);
+//            }
+//        });
+
     }
 }

--- a/full/src/test/java/apoc/custom/CypherProceduresStorageTest.java
+++ b/full/src/test/java/apoc/custom/CypherProceduresStorageTest.java
@@ -1,6 +1,7 @@
 package apoc.custom;
 
-import apoc.log.Neo4jLogStream;
+import apoc.path.PathExplorer;
+import apoc.util.FileUtils;
 import apoc.util.TestUtil;
 import org.junit.Before;
 import org.junit.Rule;
@@ -10,17 +11,14 @@ import org.neo4j.configuration.GraphDatabaseSettings;
 import org.neo4j.dbms.api.DatabaseManagementService;
 import org.neo4j.graphdb.GraphDatabaseService;
 import org.neo4j.graphdb.Node;
-import org.neo4j.internal.helpers.collection.Iterators;
-import org.neo4j.logging.AssertableLogProvider;
-import org.neo4j.logging.Log;
 import org.neo4j.test.TestDatabaseManagementServiceBuilder;
 
+import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.atomic.AtomicInteger;
 
-import static apoc.ApocConfig.apocConfig;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.jupiter.api.Assertions.assertFalse;
@@ -37,22 +35,19 @@ public class CypherProceduresStorageTest {
     private GraphDatabaseService db;
     private DatabaseManagementService databaseManagementService;
 
-    public static AssertableLogProvider logProvider = new AssertableLogProvider(true);
-
     @Before
     public void setUp() throws Exception {
-        databaseManagementService = new TestDatabaseManagementServiceBuilder(STORE_DIR.getRoot())
-                .setUserLogProvider(logProvider).build();
+        databaseManagementService = new TestDatabaseManagementServiceBuilder(STORE_DIR.getRoot()).build();
         db = databaseManagementService.database(GraphDatabaseSettings.DEFAULT_DATABASE_NAME);
-        TestUtil.registerProcedure(db, CypherProcedures.class);
+        TestUtil.registerProcedure(db, CypherProcedures.class, PathExplorer.class);
     }
 
     private void restartDb() throws IOException {
         databaseManagementService.shutdown();
         databaseManagementService = new TestDatabaseManagementServiceBuilder(STORE_DIR.getRoot()).build();
         db = databaseManagementService.database(GraphDatabaseSettings.DEFAULT_DATABASE_NAME);
-        assertTrue(db.isAvailable(60000));
-        TestUtil.registerProcedure(db, CypherProcedures.class, Neo4jLogStream.class);
+        assertTrue(db.isAvailable(1000));
+        TestUtil.registerProcedure(db, CypherProcedures.class, PathExplorer.class);
     }
     @Test
     public void registerSimpleStatement() throws Exception {
@@ -205,29 +200,49 @@ public class CypherProceduresStorageTest {
         TestUtil.testCall(db, "RETURN custom.n(2) as row", (row) -> assertEquals(2L, ((Node) row.get("row")).getProperty("value")));
     }
 
-    // TODO - METTERE LA PROCEDURA COMPLETA
     @Test
     public void testIssue1744() throws Exception {
-        db.executeTransactionally("CREATE (i:Target {value: 2});");
-        db.executeTransactionally("CALL apoc.custom.asProcedure('franco', 'RETURN \"1\" as resource', 'read', [['resource','STRING']]);");
+        db.executeTransactionally("CREATE (:Area {name: 'foo'})-[:CURRENT]->(:VantagePoint {alpha: 'beta'})");
+        db.executeTransactionally("CALL apoc.custom.asProcedure('vantagepoint_within_area',\n" +
+                "  \"MATCH (start:Area {name: $areaName} )\n" +
+                "    CALL apoc.path.expand(start,'CONTAINS>|<SEES|CURRENT','',0,100) YIELD path\n" +
+                "    UNWIND nodes(path) as node\n" +
+                "    WITH node\n" +
+                "    WHERE node:VantagePoint\n" +
+                "    RETURN DISTINCT node as resource\",\n" +
+                "  'read',\n" +
+                "  [['resource','NODE']],\n" +
+                "  [['areaName', 'STRING']],\n" +
+                "  \"Get vantage points within an area and all included areas\");");
+
+        // function analogue to procedure
+        db.executeTransactionally("CALL apoc.custom.asFunction('vantagepoint_within_area',\n" +
+                "  \"MATCH (start:Area {name: $areaName} )\n" +
+                "    CALL apoc.path.expand(start,'CONTAINS>|<SEES|CURRENT','',0,100) YIELD path\n" +
+                "    UNWIND nodes(path) as node\n" +
+                "    WITH node\n" +
+                "    WHERE node:VantagePoint\n" +
+                "    RETURN DISTINCT node as resource\",\n" +
+                "  'read',\n" +
+                "  [['areaName', 'STRING']]);");
+
+        testCallIssue1744();
         restartDb();
-        TestUtil.testCall(db, "call custom.franco", (row) -> assertEquals("1", row.get("resource")));
 
+        final String logFileContent = Files.readString(new File(FileUtils.getLogDirectory(), "debug.log").toPath());
+        assertFalse(logFileContent.contains("Could not register function: custom.vantagepoint_within_area"));
+        assertFalse(logFileContent.contains("Could not register procedure: custom.vantagepoint_within_area"));
+        testCallIssue1744();
+    }
 
-        final String serialize = logProvider.serialize();
-        assertFalse(serialize.contains("Could not register procedure"));
-//        final Log log = logProvider.getLog("");
-//         Could not register procedure:
-//        TestUtil.testResult(db, "call apoc.log.stream('neo4j.log')", (row) -> {
-//            final List<Map<String, Object>> maps = Iterators.asList(row);
-//            int intero = 0;
-//            while (row.hasNext()) {
-//                Map<String, Object> value = row.next();
-//                assertFalse(((String) value.get("line")).contains("Could not register procedure"));
-//                intero++;
-//                System.out.println("intero - " + intero);
-//            }
-//        });
+    private void testCallIssue1744() {
+        TestUtil.testCall(db, "CALL custom.vantagepoint_within_area('foo')", this::assertCallIssue1744);
+        TestUtil.testCall(db, "RETURN custom.vantagepoint_within_area('foo') as resource", this::assertCallIssue1744);
+    }
 
+    private void assertCallIssue1744(Map<String, Object> row) {
+        final Node resource = (Node) row.get("resource");
+        assertEquals("VantagePoint", resource.getLabels().iterator().next().name());
+        assertEquals("beta", resource.getProperty("alpha"));
     }
 }

--- a/full/src/test/java/apoc/custom/CypherProceduresStorageTest.java
+++ b/full/src/test/java/apoc/custom/CypherProceduresStorageTest.java
@@ -215,7 +215,7 @@ public class CypherProceduresStorageTest {
                 "  [['areaName', 'STRING']],\n" +
                 "  \"Get vantage points within an area and all included areas\");");
 
-        // function analogue to procedure
+        // function analogous to procedure
         db.executeTransactionally("CALL apoc.custom.asFunction('vantagepoint_within_area',\n" +
                 "  \"MATCH (start:Area {name: $areaName} )\n" +
                 "    CALL apoc.path.expand(start,'CONTAINS>|<SEES|CURRENT','',0,100) YIELD path\n" +

